### PR TITLE
APY + Proof of Yield from Squid

### PR DESCRIPTION
**Pull request by nick**

### CODE REVIEW

File: utils/api/fetchApyHistory.ts [hunk 1]
```diff
@@ -1,23 +1,39 @@
...
-  });
-  return data;
+async function fetchApy(days: number = 14) {
+  try {
+    const res = await fetch(process.env.NEXT_PUBLIC_SUBSQUID_URL, {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({
+        query: `query APY {
+          apies(orderBy: blockNumber_DESC, limit: ${days}, where: {timestamp_gte: "2023-06-01T00:00:00.000000Z"}) {
```
Comment (apexearth) [inline utils/api/fetchApyHistory.ts:8]: static date, is `days` even used in here?

Comment (apexearth) [reply utils/api/fetchApyHistory.ts:8]: oh i see it now, its the limit

File: hooks/analytics/useAPYChart.ts [hunk 2]
```diff
@@ -1,21 +1,46 @@
...
+  const [isFetching, setIsFetching] = useState(true);
+  const [data, setData] = useState({
+    labels: [],
+    datasets: [],
+    error: null,
   });
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const json = await fetchApyHistory(360);
```
Comment (apexearth) [inline hooks/analytics/useAPYChart.ts:19]: no way or reason to refine how many days we pull?

File: queries/useApyHistoryQuery.ts [hunk 3]
```diff
@@ -1,43 +1,10 @@
...
-  data?.["apy7"]?.pop();
-  data?.["apy30"]?.pop();
-
-  return data;
-}
+import { fetchApyHistory } from "../utils";
 
 const useApyHistoryQuery = (apyHistory, options = {}) => {
-  return useQuery(QUERY_KEYS.ApyHistory(365), fetchApyHistory, {
+  return useQuery(QUERY_KEYS.ApyHistory(365), () => fetchApyHistory(), {
```
Comment (apexearth) [inline queries/useApyHistoryQuery.ts:7]: What effect does this change have?

File: utils/api/fetchProofOfYieldByDay.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            tx_hash: event.txHash,
+            block_number: event.blockNumber,
+            block_time: event.timestamp,
+          };
+        }),
+      },
+    ];
+  } catch (err) {
+    console.log(`Failed to fetch daily stats: ${err}`);
+  }
```
Comment (apexearth) [inline utils/api/fetchProofOfYieldByDay.ts:82]: What happens when we error in these? How does the front-end react? It seems like a bad idea to silently fail. I wouldn't do this. I'd put any silence on failures in the view component.

File: utils/api/fetchApy.ts [hunk 5]
```diff
@@ -1,19 +1,29 @@
...
+        }`,
+        variables: null,
+        operationName: "APY",
+      }),
+    });
+    const json = await res.json();
+
+    return [
+      json.data.apies[0].apy7DayAvg / 100,
+      json.data.apies[0].apy30DayAvg / 100,
```
Comment (apexearth) [inline utils/api/fetchApy.ts:22]: Sad to say it but the resolver should be returning these values in the correct percentages.  We should not have to do this. The data squid gives us is bad. We should fix that.

Comment (apexearth) [reply utils/api/fetchApy.ts:22]: 7.5321496604432046 -> 0.075321496604432046

Comment (apexearth) [reply utils/api/fetchApy.ts:22]: Since it is a resolver change (I think) we should be able to maintain v6 I'm guessing.

**Review by apexearth — CHANGES_REQUESTED** [review]
I think the APY coming out of subsquid should be changed 7.5321496604432046 -> 0.075321496604432046.

**Review by apexearth — APPROVED** [review]
ok, i see you didn't do that with the APY, and it was originally created like that.
so good as is... left a few comments thats all

File: utils/api/fetchApy.ts (hunk 5 above)
Comment (apexearth) [reply utils/api/fetchApy.ts:22]: nevermind, not your fault

File: queries/useApyHistoryQuery.ts (hunk 3 above)
Comment (nick) [reply queries/useApyHistoryQuery.ts:7]: it was a typescript thing. I guess useQuery expects the fetch function to follow a certain convention

File: utils/api/fetchProofOfYieldByDay.ts (hunk 4 above)
Comment (nick) [reply utils/api/fetchProofOfYieldByDay.ts:82]: the old function was in this format, I just changed the data fetching piece. I think there are a lot of things like this but I didn't want to start refactoring before we decide the longer term future of the marketing sites

File: hooks/analytics/useAPYChart.ts (hunk 2 above)
Comment (nick) [reply hooks/analytics/useAPYChart.ts:19]: Not sure I understand... the 360 arg is the number of days to pull?

Comment (apexearth) [reply hooks/analytics/useAPYChart.ts:19]: i guess this chart is always just the same # of days? was wondering why we don't have the number of days in a param or something

**Pull request closed without merging**